algorithms/kernel/dtrees/gbt/gbt_train_aux.i

```diff
@@ -61,6 +61,13 @@ void deleteTables(gbt::internal::GbtDecisionTree** aTbl, HomogenNumericTable<dou
     }
 }
 
+template <CpuType cpu>
+size_t getThrOptBorder(size_t nThreads)
+{
+    // TODO: investigate effect of threading optimization on different cpus and numbers of threads
```

Comment:
Seems this value depends on cache sizes. Similar techniques can be applied here as for TileDimensions class.